vms/avm/txs/fees/calculator.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fees
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/vms/avm/config"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+)
+
+var _ txs.Visitor = (*Calculator)(nil)
+
+type Calculator struct {
+	// setup, to be filled before visitor methods are called
+	Config    *config.Config
+	ChainTime time.Time
```

Comment:
while ChainTime is not strictly necessary here, we'll use it to check the active fork in next PRs. We keep ChainTime here so that we initialize it in this PR and we reduce diffs down the line